core-to-plc/src/Language/Plutus/CoreToPLC/Plugin.hs

```diff
@@ -75,6 +75,14 @@ instance FromJSON PlcCode where
       Left e  -> fail e
       Right v -> pure v
 
+lifted :: LiftPlc a => a -> PlcCode
```

Comment:
I'm not wild about this, and here's I think why. `PlcCode` is really supposed to be the opaque type produced by the plugin. As such I want to expose as few operations on it as possible, since otherwise it's harder to change.

The upshot of this is that we perhaps shouldn't use it as the type of code on the chain. We might use another type for that (probably isomorphic since it's going to also contain serialized code) but crucially I can change `PlcCode` without affecting that.

Possibly nothing to do here, maybe I'll do something in a follow-up.

Reply:
I agree, it makes sense to split up the two uses of `PlcCode`, but changing that should happen in a different PR (adding `lifted` is just a consequence of the fact that `PlcCode` is also used as the on-chain type right now)

The operations we need on `PlcCode` _after_ it has been put on the chain are

* `apply` to other `PlcCode`s and values that can be `lift`ed
* `evaluate`

So maybe we need a `newtype OnChainCode = OnChainCode ByteString` with `apply`, `evaluate`, `lifted` and `PlcCode -> OnChainCode` ?

Reply:
Exactly! It could probably even be a `SerializedProgram` type in `language-plutus-core`.